### DISCUSSION TITLE
Fish history format

### DISCUSSION THREAD

**Discussion by lnicola** [fish-shell/fish-shell#9514] [Q&A] [answered]
I'd like to give fish another try, but I want to import my existing shell history (from ZSH). I found a script, but it doesn't work very well because I don't have the timestamps (and it doesn't escape the output anyway), so I wanted to rewrite or update it.

fish uses something similar to, but not quite YAML (see also https://github.com/fish-shell/fish-shell/issues/4675), so I'm not sure what to output. It appears to `\`-escape some special characters, including `\`, but is there more precise description of the format?

**Comment by krobelus** [comment, marked as answer]
only two characters are escaped:

```c++
// Support for escaping and unescaping the nonstandard "yaml" format introduced in fish 2.0.
static void escape_yaml_fish_2_0(std::string *str) {
    replace_all(str, "\\", "\\\\");  // replace one backslash with two
    replace_all(str, "\n", "\\n");   // replace newline with backslash + literal n
}
```
